src/domain/wallets/index.types.d.ts

```diff
@@ -12,11 +10,47 @@ type Deprecated = {
   readonly feeUsd: number
 }
 
+type InitiationViaIntraledger = {
+  readonly type: PaymentInitiationMethod["IntraLedger"]
+  readonly walletId: WalletId
+  readonly counterPartyUsername: Username
+}
+
+type InitiationViaLn = {
+  readonly type: PaymentInitiationMethod["Lightning"]
+  readonly paymentHash: PaymentHash
+  readonly pubkey: Pubkey
+}
+
+type InitiationViaOnChain = {
+  readonly type: PaymentInitiationMethod["OnChain"]
+  readonly address: OnChainAddress
+}
+
+type InitiationViaOnChainLegacy = {
```

Comment:
why is that needed?

Reply:
This was for onchain transaction that have `undefined` address in the db for older version of the codebase that didn't persist this